### PR TITLE
Test daily guideline checks

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Adds a test to help flag potential bugs in `fct_daily_guideline_checks`; tests that checks can only come from one "source" and that the entities match up as we expect. For example, a check for schedule feeds ends up living on a schedule feed entity grained record, etc.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
dbt runs and passes on the changed tables

### CODE REVIEW

File: warehouse/models/mart/gtfs_quality/fct_daily_guideline_checks.sql [hunk 1]
```diff
@@ -66,13 +66,27 @@ fct_daily_guideline_checks AS (
...
-        ) AS status
+        ) AS status,
+        CASE
+            WHEN schedule_feed_checks.status IS NOT NULL THEN idx.entity = {{ schedule_feed() }}
+            WHEN rt_feed_checks.status IS NOT NULL THEN idx.entity = {{ rt_feed() }}
+            WHEN schedule_url_checks.status IS NOT NULL THEN idx.entity = {{ schedule_url() }}
+            WHEN rt_url_checks.status IS NOT NULL THEN idx.entity = {{ rt_url() }}
+            WHEN organization_checks.status IS NOT NULL THEN idx.entity = {{ organization() }}
+        END AS matches_entity,
+        CASE WHEN schedule_feed_checks.status = 'PASS' THEN 1 ELSE 0 END
```
Comment (owades) [inline warehouse/models/mart/gtfs_quality/fct_daily_guideline_checks.sql:84]: shouldn't this be `WHEN schedule_feed_checks.status IS NOT null THEN 1 ELSE 0 `?

**Review by owades — APPROVED** [review]
I left one question, guessing you have a good answer but perhaps it needs to be changed.

File: warehouse/macros/define_gtfs_guidelines.sql [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+{%- endmacro %}
+
+{% macro rt_url() -%}
+'rt_url'
+{%- endmacro %}
+
+{% macro organization() -%}
+'organization'
+{%- endmacro %}
+
```
Comment (lauriemerrell) [inline warehouse/macros/define_gtfs_guidelines.sql:23]: Between you and @owades what is the plan for adding "service"? Wondering if @owades needs to do that on #2206 or not

File: warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml [hunk 3]
```diff
@@ -207,6 +212,9 @@ models:
       - *check
       - name: status
       - name: feature
+      - name: matches_entity
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml:215]: If this is being surfaced on this mart table can you add a short description indicating that this is just for testing/auditing? Same for below

**Review by lauriemerrell — COMMENTED** [review]
Just two logistical/docs questions. Thanks for tackling this

File: warehouse/macros/define_gtfs_guidelines.sql (hunk 2 above)
Comment (atvaccaro) [reply warehouse/macros/define_gtfs_guidelines.sql:23]: I'll add the macros here; then @owades just has to use them in intended_checks

File: warehouse/macros/define_gtfs_guidelines.sql [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+{% macro organization() -%}
```
Comment (owades) [inline warehouse/macros/define_gtfs_guidelines.sql:24]: fix macro names (id "service" macro is called "organization()")

Comment (atvaccaro) [reply warehouse/macros/define_gtfs_guidelines.sql:24]: 🤦 thank you

File: warehouse/macros/define_gtfs_guidelines.sql [hunk 5]
```diff
@@ -21,15 +21,15 @@
...
 'service'
 {%- endmacro %}
 
-{% macro organization() -%}
+{% macro gtfs_service_data() -%}
 'gtfs_service_data'
 {%- endmacro %}
 
-{% macro organization() -%}
+{% macro gtfs_dataset() -%}
```
Comment (owades) [inline warehouse/macros/define_gtfs_guidelines.sql:32]: there's also those dash symbols

Comment (atvaccaro) [reply warehouse/macros/define_gtfs_guidelines.sql:32]: yes that's for stripping whitespace

Comment (owades) [reply warehouse/macros/define_gtfs_guidelines.sql:32]: now I know!

**Review by lauriemerrell — APPROVED** [review]
👍 I do think maybe a *slightly* more verbose description of the testing fields wouldn't hurt, but it is fine / not blocking IMO

**Pull request merged**